resolwe/flow/executors/__init__.py

```diff
@@ -122,6 +123,14 @@ def run(self, data_id, script, verbosity=1):
 
         self.data_id = data_id
 
+        # Fetch data instance to get any executor requirements.
+        if data_id != 'no_data_id':
```

Comment:
I have created a `with_docker_executor` decorator, which handles setting up the Docker executor for a specific unit test. It is now used both for testing if Docker executor works and for testing if specifying Docker requirements works.